File: functions/libgcrypt/variants/go/0_barrett_reduce_8.go
package main

func barrettReduce(a int16) int16 {
	const v = int16(((1 << 26) + 3329/2) / 3329)
	t := ((int32(a) * int32(v)) + (1 << 25)) >> 26
	return int16(int32(a) - t*3329)
}


func main(){
    barrettReduce(0)
}

File: functions/libgcrypt/variants/c/0_barrett_reduce_8.c
#include <stdint.h>

int16_t barrett_reduce(int16_t a) {
  int32_t t;
  const int16_t v = 12482;

  t = (((int32_t)v * a + (1L << 25)) >> 26) * 3329;
  return a - (int16_t)t;
}


int main(){
    barrett_reduce(0);
}
